packages/trimerge-sync-indexed-db/src/trimerge-indexed-db.ts

```diff
@@ -269,11 +317,22 @@ class IndexedDbBackend<
             if (await commitExistsAlready(commit)) {
               return;
             }
-            await commitsDb.add({ syncId, remoteSyncId: '', ...commit });
-            const ackedCommit = await commitsDb.get(commit.ref);
-            if (ackedCommit) {
-              refs.set(ref, ackedCommit.main);
+
+            if (this.getStoreCommitMetadata) {
+              commit.metadata = {
+                ...commit.metadata,
+                ...this.getStoreCommitMetadata(
```

Comment:
I guess my thinking about this is it feels like an in between having something at the top `TrimergeClient` level where people can have hooks to arbitrary points in the commit lifecycle, and just explicitly doing the metadata modification here. I'm thinking maybe we should just inline the metadata modification here and if we feel the need to parameterize it in the future we add this later?